tests/Adapters/Microsoft.Bot.Builder.Adapters.Facebook.Tests/FacebookAdapterTests.cs

```diff
@@ -0,0 +1,189 @@
+
+        [Fact]
+        public void ConstructorShouldFailWithNullClient()
+        {
+            Assert.Throws<ArgumentNullException>(() => { new FacebookAdapter((FacebookClientWrapper)null); });
+        }
+
+        [Fact]
+        public void ConstructorWithArgumentsShouldSucceed()
+        {
+            Assert.NotNull(new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object));
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldFailWithNullConversationReference()
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+
+            Task BotsLogic(ITurnContext turnContext, CancellationToken cancellationToken)
+            {
+                return Task.CompletedTask;
+            }
+
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => { await facebookAdapter.ContinueConversationAsync(null, BotsLogic); });
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldFailWithNullLogic()
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var conversationReference = new ConversationReference();
+
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => { await facebookAdapter.ContinueConversationAsync(conversationReference, null); });
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldSucceed()
+        {
+            var callbackInvoked = false;
+
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var conversationReference = new ConversationReference();
+            Task BotsLogic(ITurnContext turnContext, CancellationToken cancellationToken)
+            {
+                callbackInvoked = true;
+                return Task.CompletedTask;
+            }
+
+            await facebookAdapter.ContinueConversationAsync(conversationReference, BotsLogic);
+            Assert.True(callbackInvoked);
+        }
+
+        [Fact]
+        public async System.Threading.Tasks.Task DeleteActivityAsyncShouldThrowNotImplementedException()
+        {
+            var adapter = new FacebookAdapter(new FacebookClientWrapper(_testOptions));
+
+            var activity = new Activity();
+            var turnContext = new TurnContext(adapter, activity);
+
+            var conversationReference = new ConversationReference();
+
+            await Assert.ThrowsAsync<NotImplementedException>(() => adapter.DeleteActivityAsync(turnContext, conversationReference, default));
+        }
+
+        [Fact]
+        public async void ProcessAsyncShouldThrowExceptionWithInvalidBody()
+        {
+            var clientWrapper = new Mock<FacebookClientWrapper>(_testOptions);
+            clientWrapper.Setup(x => x.VerifySignature(It.IsAny<HttpRequest>(), It.IsAny<string>())).Returns(true);
```

Comment:
What's `x`? Any change we get a more expressive name? Not sure about the convention here, this might be fine